src/Form/formatters.js

```diff
@@ -104,10 +104,125 @@ const getCountryName = (el, options = {}) => {
+
+    const localisedCountryNamesToCodes = getLocalisedCountryNamesToCodes(el)
+    if (localisedCountryNamesToCodes[el.value]) return localisedCountryNamesToCodes[el.value]
+
+    if (el instanceof HTMLSelectElement) {
+        const selectedText = el.selectedOptions[0].text
+        if (COUNTRY_CODES_TO_NAMES[selectedText]) return selectedText
+        if (COUNTRY_NAMES_TO_CODES[selectedText]) return localisedCountryNamesToCodes[selectedText]
+        if (localisedCountryNamesToCodes[selectedText]) return localisedCountryNamesToCodes[selectedText]
+    }
+    return ''
+}
+
+/** @param {InternalDataStorageObject} credentials */
+const shouldStoreCredentials = ({credentials}) =>
+    credentials.password
+
+/** @param {InternalDataStorageObject} credentials */
+const shouldStoreIdentities = ({identities}) =>
+    (identities.firstName || identities.fullName) && identities.addressStreet && identities.addressCity
+
+/** @param {InternalDataStorageObject} credentials */
+const shouldStoreCreditCards = ({creditCards}) =>
+    creditCards.cardNumber && creditCards.cardSecurityCode
+
+/**
+ * Formats form data into an object to send to the device for storage
+ * If values are insufficient for a complete entry, they are discarded
+ * @param {InternalDataStorageObject} formValues
+ * @return {DataStorageObject}
+ */
+const prepareFormValuesForStorage = (formValues) => {
+    let {credentials, identities, creditCards} = formValues
+
+    /** Fixes for credentials **/
+    // Don't store if there isn't enough data
+    if (shouldStoreCredentials(formValues)) {
+        // If we don't have a username to match a password, let's see if the email is available
+        if (credentials.password && !credentials.username && identities.emailAddress) {
+            credentials.username = identities.emailAddress
+        }
+    } else {
+        // @ts-ignore
+        credentials = null
+    }
+
+    /** Fixes for identities **/
+    // Don't store if there isn't enough data
+    if (shouldStoreIdentities(formValues)) {
+        if (identities.fullName) {
+            // If the fullname can be easily split into two, we'll store it as first and last
+            const nameParts = identities.fullName.trim().split(/\s+/)
+            if (nameParts.length === 2) {
+                identities.firstName = nameParts[0]
+                identities.lastName = nameParts[1]
+            } else {
+                // If we can't split it, just store it as first name
+                identities.firstName = identities.fullName
+            }
+            delete identities.fullName
+        }
+    } else {
+        // @ts-ignore
+        identities = null
+    }
+
+    /** Fixes for credit cards **/
+    // Don't store if there isn't enough data
+    if (shouldStoreCreditCards(formValues)) {
+        if (creditCards.expiration) {
+            const [expirationMonth, expirationYear] = creditCards.expiration.split(/\D/)
```

Comment:
We could get smarter with this, but I think this is good enough for this iteration. Let me know if you disagree.

Reply:
to comment on this, I'd need to have some examples of values that `creditCards.expiration` might have?

Reply:
There is the possibility of `YYYY / MM`, but it's definitely rare. We can follow up or I can add more logic if I have some downtime today.